sdk/src/main/java/co/omise/android/ui/FpxEmailFormFragment.kt

```diff
@@ -0,0 +1,72 @@
+package co.omise.android.ui
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.INVISIBLE
+import android.view.ViewGroup
+import android.widget.Button
+import co.omise.android.R
+import co.omise.android.extensions.setOnAfterTextChangeListener
+import co.omise.android.extensions.setOnClickListener
+import co.omise.android.models.Bank
+import co.omise.android.models.Capability
+import co.omise.android.models.PaymentMethod
+import co.omise.android.models.Source
+import kotlinx.android.synthetic.main.fragment_fpx_email_form.*
+
+/**
+ * FpxEmailFormFragment is the UI class to show an email form for FPX payments.
+ */
+class FpxEmailFormFragment : OmiseFragment() {
+
+    var navigation: PaymentCreatorNavigation? = null
+    var requester: PaymentCreatorRequester<Source>? = null
+
+    private val emailEdit: OmiseEditText by lazy { edit_email }
+    private val submitButton: Button by lazy { button_submit }
+    private val allowedEmailFormat = "\\A[\\w+\\-.]+@[a-z\\d\\-.]+\\.[a-z]{2,}\\z"
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_fpx_email_form, container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        title = getString(R.string.payment_method_fpx_title)
+        setHasOptionsMenu(true)
+
+        with(emailEdit) {
+            setOnAfterTextChangeListener(::updateSubmitButton)
+        }
+
+        submitButton.setOnClickListener(::submitForm)
+    }
+
+    private fun updateSubmitButton() {
+        val text = emailEdit.text.toString()
+        when {
+            text.isEmpty() || allowedEmailFormat.toRegex().matches(text) -> submitButton.isEnabled = true
+            else -> submitButton.isEnabled = false
+        }
+    }
+
+    private fun submitForm() {
+        val requester = requester ?: return
+        val email = emailEdit.text?.toString()?.trim().orEmpty()
+        val banks = requester.capability.paymentMethods?.find { it.name.equals("fpx") }?.banks ?: mockBanks()
+
+        navigation?.navigateToFpxBankChooser(banks, email)
+    }
+
+    private fun mockBanks() : List<Bank> {
```

Comment:
What is the purpose of these mocks? Usually we don't want them in production code

Reply:
<img width="300" alt="Screen Shot 2564-03-05 at 09 43 52" src="https://user-images.githubusercontent.com/61586978/110059869-a95e5f00-7d97-11eb-9d9c-b41d9954fe20.png">

These are sample banks that would appear in test mode (to mimic the responses from Capability API).

Reply:
https://github.com/omise/omise-android/pull/192/files#r588019130

Reply:
@hpariwat Basically, the `Specify Payment Methods` will create a capability object and put it to the SDK. In case that you want to test with the sample with test banks, the SDK no need to know about it used `Specify Payment Methods` or not. :)